kv: remove server-side support of InitPuts

yuzefovich opened the pull request:
We removed client-side usage and support of InitPut requests in 46787b3399ea5c6e20cebf0566d4cf7be4356345 which merged during 25.1 cycle. Given that we no longer need to maintain compatibility with 24.3 versions, we can remove server-side support as well.

Fixes: #71074.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146855)
<!-- Reviewable:end -->

yuzefovich committed:
kv: remove server-side support of InitPuts

We removed client-side usage and support of InitPut requests in
46787b3399ea5c6e20cebf0566d4cf7be4356345 which merged during 25.1 cycle.
Given that we no longer need to maintain compatibility with 24.3
versions, we can remove server-side support as well.

Release note: None

RaduBerinde reviewed (commented):
Nice! pkg/storage changes LGTM

A bit surprised no tests are failing since we haven't bumped the min supported version yet.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146855#-:-OQkTYZ727L_QzV2vbtQ:b-t6a41o)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @arulajmani and @stevendanna)



<!-- Sent from Reviewable.io -->

stevendanna reviewed (approved):
Thanks for doing this cleanup!

yuzefovich commented:
TFTRs!

bors r+
